.github/workflows/iwyu.yaml

```diff
@@ -80,7 +81,16 @@ jobs:
           done
       - run: |
           echo "::remove-matcher owner=clang-include-cleaner::"
+      - run: |
+          if git -c safe.directory="$PWD" grep -E '#include +"seastar/' > "$SEASTAR_BAD_INCLUDE_OUTPUT_PATH"; then
+            echo "ERROR: Found #include \"seastar/ in the source code. Use angle brackets instead."
+            exit 1
+          fi
       - uses: actions/upload-artifact@v4
         with:
           name: Logs (clang-include-cleaner)
           path: "./${{ env.CLEANER_OUTPUT_PATH }}"
+      - uses: actions/upload-artifact@v4
+        with:
+          name: Logs (seastar-bad-include)
+          path: "./${{ env.SEASTAR_BAD_INCLUDE_OUTPUT_PATH }}"
```

Comment:
@avikivity i made some improvements on top of your change at #23310

- add inline annotations using problem matchers, see https://github.com/actions/toolkit/blob/main/docs/problem-matchers.md
- use a single step for uploading both output files, because the `path` setting is actually passed to [@actions/glob](https://github.com/actions/toolkit/tree/main/packages/glob), i removed the double quotes and the leading "./" from the paths.
- use "::error" workflow command to signify the failure, see https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#example-creating-an-annotation-for-an-error